### PR TITLE
Dataset individual page

### PR DESCRIPTION
**Pull request by luccasmmg**

### CODE REVIEW

**Comment by Gutts-n** [comment]
@luccasmmg I opened the page without signing in and received this error 

image

Link: https://tdc-data-portal-git-dataset-individual-page-datopian1.vercel.app/Transport%20Data%20Commons/c4a9f489-1d6a-4760-9097-b5866604feb1

File: frontend/src/server/api/routers/user.ts [hunk 1]
```diff
@@ -143,11 +143,11 @@ export const userRouter = createTRPCRouter({
...
     }
   }),
-  getUsersByIds: protectedProcedure
+  getUsersByIds: publicProcedure
     .input(z.object({ ids: z.array(z.string()) }))
     .query(async ({ input, ctx }) => {
-      const user = ctx.session.user;
-      const apiKey = user.apikey;
+      const user = ctx.session?.user;
+      const apiKey = user?.apikey ?? env.SYS_ADMIN_API_KEY;
```
Comment (Gutts-n) [inline frontend/src/server/api/routers/user.ts:150]: Shouldn't you send an empty key when the user is not signed in to return the public data from the CKAN?

File: frontend/src/components/dataset/individualPage/Downloads.tsx [hunk 2]
```diff
@@ -1,116 +1,46 @@
...
+  const { data: datasetDownloads, isLoading } =
+    api.ga.getDownloadStats.useQuery({
+      id: dataset.id,
+    });
+  const dataResources = dataset.resources.filter(
+    (r) => r.resource_type && r.resource_type === "data"
+  );
+  const docsResources = dataset.resources.filter(
+    (r) => r.resource_type && r.resource_type === "documentation"
+  );
```
[the remark this answers is not in the document]
Comment (luccasmmg) [reply frontend/src/components/dataset/individualPage/Downloads.tsx:22]: Thats very ugly code what you are suggesting

File: frontend/src/server/api/routers/user.ts (hunk 1 above)
Comment (luccasmmg) [reply frontend/src/server/api/routers/user.ts:150]: thanks for spotting this, this is no longer needed cause of the backed changes

**Comment by luccasmmg** [comment]
> @luccasmmg I opened the page without signing in and received this error
> 
> image
> 
> Link: https://tdc-data-portal-git-dataset-individual-page-datopian1.vercel.app/Transport%20Data%20Commons/c4a9f489-1d6a-4760-9097-b5866604feb1

Good catch, fixed, there were two errors

One in the `dataset tab` which should not appear when the dataset doesnt have any datastore_active resources, so i fixed that

The other is in the `metadata` tab, which technically happened because the live deployment doesnt have the updates to `package_show` which i added, nonetheless i added error handling to take care of it, even tough i do not expect the value which was undefined here to ever be undefined once this PR gets merged

File: frontend/src/components/dataset/individualPage/Downloads.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gutts-n) [reply frontend/src/components/dataset/individualPage/Downloads.tsx:22]: It's ugly but you won't run through the array twice.

**Pull request merged**